examples/patterns/instantiate/structured_config/my_app.py

```diff
@@ -0,0 +1,103 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+from dataclasses import dataclass, field
+from typing import Any, List
+
+from omegaconf import MISSING, DictConfig
+
+import hydra
+from hydra.core.config_store import ConfigStore
+from hydra.types import ObjectConf
+
+
+class DBConnection:
+    def connect(self) -> None:
+        pass
+
+
+class MySQLConnection(DBConnection):
+    def __init__(self, host: str, port: int, user: str, password: str) -> None:
+        self.driver = "MySQL"
+        self.host = host
+        self.port = port
+        self.user = user
+        self.password = password
```

Comment:
pull up common fields into DBConnection.
you can can the example on [this](https://github.com/facebookresearch/hydra/blob/master/examples/tutorials/structured_configs/3_config_groups/my_app_with_inheritance.py), it's more compact and to the point.